### PR TITLE
Check for invalid byte offsets before loading/scanning

### LINKED ISSUE #64: catch bad segyio Enum prior to loading/scanning
If the Enum is not valid for SEGYIO then it should be caught and an error raised prior to scanning to catch user input errors.

For example, here 194 is a typo and should be 193

```
segy.segy_loader(template, iline=189, xline=194)
```

### PR DESCRIPTION
**Pull request by stpotter16**
Closes #64 

This PR adds a sanity checking function `_validate_offset_val` that checks the key byte offset values passed to `segysak.segy._segy_loader:_loader_converter_header_handling` for invalid `segyio Enum` values and raises a `ValueError` if found.

### CODE REVIEW

File: segysak/segy/_segy_loader.py [hunk 1]
```diff
@@ -35,6 +35,8 @@
     petrel_2d=dict(cdp=21, cdpx=73, cdpy=77),
 )
 
+UNRECOGNIZED_OFFSET = 'Unknown Enum'
+
```
Comment (stpotter16) [inline segysak/segy/_segy_loader.py:39]: Defining this module level constant is potentially YAGNI (you aren't going to need it) as it's only used in a single place in this module. My rationale for doing things this way was to semi-future-proof things a little bit. We're relying on `segyio`'s error message for a bad byte offset value to always be `Unkown Enum`, which is a fragile assumption and could break in newer versions of `segyio`

**Review by stpotter16 — COMMENTED** [review]
@trhallam added some design comments from my perspective. Your feedback welcome!

**Comment by trhallam** [comment]
Aw man, I wish I had seen this sooner, I was literally just working on this. 

https://github.com/trhallam/segysak/commit/8957ca635fc193d9227ccd0c96f2cfd115137fe3

I went a slightly different route where I check against the keys. in the `segyio.tracefield.keys` object.

I don't like that either because it requires the keys and enum to be synced but I don't think there are any other options because segyio doesn't use the default Enum class from Python.

What do you think?

**Comment by stpotter16** [comment]
>  I was literally just working on this.

haha, I'm a bit late to the hackathon this week, so no worries!

Regarding your solution -- it's more robust, I think, than what I have (and is applied to other parts of the code I overlooked, e.g. `_segy_writer`).

I agree that it's a little unfortunate to rely on syncing between the `segyio.tracefield.keys` and the underlying enum, but I think it's more robust to check the `keys` data structure explicitly than rely on a stringy error message like I have.

I think I'll go ahead and close my PR. Cool project though; keep up the good work!

**Comment by stpotter16** [comment]
Solution in https://github.com/trhallam/segysak/commit/8957ca635fc193d9227ccd0c96f2cfd115137fe3 preferred. Closing here.

**Comment by trhallam** [comment]
Thanks for your input anyway! If you think of anything you want added or can improve, feel free to have a go.

**Pull request closed without merging**